### PR TITLE
Feat/customize completed stepper state

### PR DESCRIPTION
**Pull request by 0xLodz**
## Purpose
Allow to customize a completed stepper.

Once stepper is completed, a customized message can be displayed close to the progress bar, and actions buttons can be hidden.

image

Customization is available through the optional props `completedMessage` and `hideActionsOnComplete`.

## Testing
Update the `shareDataset.tsx` file, by adding these props to the stepper component:
```tsx
import type { FC } from 'react'
import { useTranslation } from 'react-i18next'
import * as O from 'fp-ts/Option'
import { ServiceStorageSelection } from './steps/serviceStorageSelection/serviceStorageSelection'
import { DataSelection } from './steps/dataSelection/dataSelection'
import { Summary } from './steps/summary/summary'
import { MetadataFilling } from './steps/metadataFilling/metadataFilling'
import { Stepper } from '@/ui/component/stepper/stepper'
import type { StepElement } from '@/ui/component/stepper/stepper'
import '../i18n/index'
import './shareDataset.scss'
import { useAppStore, useFileStore } from '@/ui/store'
import { CompleteTransaction } from './steps/completeTransaction/completeTransaction'

export const ShareDataset: FC = () => {
  const { t } = useTranslation('share')
  const { storageServiceId } = useAppStore(store => ({
    storageServiceId: store.shareData.storageServiceId
  }))
  const { hasStoredFiles } = useFileStore(store => ({
    hasStoredFiles: store.hasStoredFiles
  }))

  const storageServiceSelection: StepElement = {
    id: 'step1',
    content: <ServiceStorageSelection />,
    validate: () => O.isSome(storageServiceId)
  }
  const dataSelection: StepElement = {
    id: 'step2',
    content: <DataSelection />,
    validate: () => hasStoredFiles()()
  }
  const metadataFilling: StepElement = { id: 'step3', content: <MetadataFilling /> }
  const summary: StepElement = { id: 'step4', content: <Summary /> }
  const completeTransaction: StepElement = { id: 'step5', content: <CompleteTransaction /> }

  const steps = [
    storageServiceSelection,
    dataSelection,
    metadataFilling,
    summary,
    completeTransaction
  ]

  return (
    <div className="okp4-dataverse-portal-share-dataset-page-main">
      <h1>{t('share.dataset.title')}</h1>
      <Stepper completedMessage="Done!" hideActionsOnComplete steps={steps} />
    </div>
  )
}
```

## Demo
https://www.loom.com/share/eb052e9a79f7452c977ba1903e93e5bb?sid=81fa725b-e476-4f95-87ab-8f4fb07b4b12

### CODE REVIEW

File: src/ui/component/stepper/stepper.tsx [hunk 1]
```diff
@@ -29,10 +35,15 @@ export const Stepper: FC<StepperProps> = ({ steps: stepElements }) => {
   const activeStepElement = stepElements.find(step => step.id === activeStepId) ?? stepElements[0]
   const activeStep = steps.find(step => step.id === activeStepId) ?? steps[0]
 
+  const isEveryStepValid = stepElements.every(stepElement => stepElement.validate?.() ?? true)
+  const isLastStep = activeStep.order === steps.length - 1
```
Comment (lolottetheclash) [inline src/ui/component/stepper/stepper.tsx:39]: I think this constants should be calculated and returned by the hook to reduce code duplication when using the stepper in several places, what do you think?

**Review by lolottetheclash — CHANGES_REQUESTED** [review]
well done!
just a remark abut the stepper's hook !

File: src/ui/component/stepper/stepper.tsx (hunk 1 above)
Comment (0xLodz) [reply src/ui/component/stepper/stepper.tsx:39]: The result would be the same because a new hook function is created for each Stepper instance.

Comment (lolottetheclash) [reply src/ui/component/stepper/stepper.tsx:39]: For me it's the same logiq as getting back `activeStepId` or `previousActiveStepId` from the hook, your new 
 constants could be calculated in the hook and be returned by it. @fredericvilcot any advice on it?

Comment (0xLodz) [reply src/ui/component/stepper/stepper.tsx:39]: Also, we currently have no references to `stepElement` within the hook, so we cannot call the validate functions from the hook.

Comment (ccamel) [reply src/ui/component/stepper/stepper.tsx:39]: AFAIS, the hook should be improved to support step validation. I can also see a few other areas for improvement, but we'll see at a later date.

Comment (lolottetheclash) [reply src/ui/component/stepper/stepper.tsx:39]: ok so let's go :)

**Review by lolottetheclash — APPROVED** [review]
👍

**Review by Yoj16 — APPROVED** [review]
Nice work 😉

**Pull request merged**